htdocs/mri/jiv/get_file.php

```diff
@@ -80,76 +73,65 @@
 }
 unset($path_parts);
 
-// Make sure that the user isn't trying to break out of the $path by
-// using a relative filename.
-// No need to check for '/' since all downloads are relative to $imagePath,
-// $DownloadPath or $mincPath
-if (strpos("..", $File) !== false) {
-    error_log("ERROR: Invalid filename");
-    header("HTTP/1.1 400 Bad Request");
-    exit(4);
-}
+/* Prevent directory traversal by resolving the path first */
+$baseFile = basename(realpath($File));
 
 switch($FileExt) {
 case 'mnc':
-    $FullPath         = $mincPath . '/' . $File;
-    $MimeType         = "application/x-minc";
-    $DownloadFilename = basename($File);
+    $FullPath = $mincPath . '/' . $baseFile;
+    $MimeType = "application/x-minc";
     break;
 case 'nii':
-    $FullPath         = $mincPath . '/' . $File;
-    $MimeType         = "application/x-nifti";
-    $DownloadFilename = basename($File);
+    $FullPath = $mincPath . '/' . $baseFile;
+    $MimeType = "application/x-nifti";
     break;
 case 'nii.gz':
-    $FullPath         = $mincPath . '/' . $File;
-    $MimeType         = "application/x-nifti-gz";
-    $DownloadFilename = basename($File);
+    $FullPath = $mincPath . '/' . $baseFile;
+    $MimeType = "application/x-nifti-gz";
     break;
 case 'png':
-    $FullPath = $imagePath . '/' . $File;
+    $FullPath = $imagePath . '/' . $baseFile;
     $MimeType = "image/png";
     break;
 case 'jpg':
-    $FullPath = $imagePath . '/' . $File;
+    $FullPath = $imagePath . '/' . $baseFile;
```

Comment:
Is `$DownloadedFilename` available somewhere outside of this script? The only place I see it used here is in the `Content-Disposition` header below and in that case I have preserved the value as in all cases `$DownloadedFilename = basename($File);`.